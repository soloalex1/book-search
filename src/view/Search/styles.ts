import styled from "styled-components";

export const Container = styled.div`
  width: 100%;
`;

export const Content = styled.div`
  width: 100%;
  display: grid;
  grid-template-columns: 1fr 5fr;
`;

export const ContentResults = styled.div`
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  gap: 24px;
  margin: 49px 0;
  width: 100%;
`;

export const ContentResultsWrapper = styled.div`
  margin: 10px 0;
`;

export const ContentResultsCover = styled.div`
  img {
    width: 124px;
    height: 185px;
  }
`;

export const ContentResultsTitle = styled.div`
  label {
    font-size: 14px;
    color: #86878b;
  }
`;

export const ContentResultsCategory = styled.div`
  span {
    font-size: 14px;
    color: #9eaeb7;
  }
`;
