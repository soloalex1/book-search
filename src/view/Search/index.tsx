import React, { useEffect } from "react";
import Filters from "../../components/filters";
import Footer from "../../components/footer";
import {
  footerAllrightsReserved,
  helpLink,
  privacyAndPolicy,
  termsAndUsage,
} from "../../constants";

import useStore from "../../store";

import * as S from "./styles";
import { VolumeInfo } from "../../types";
import Spinner from "../../components/spinner";

const Search: React.FC = () => {
  const { volumes, setSuggestions, setLoading } = useStore((state) => state);

  useEffect(() => {
    setSuggestions([]);
    setLoading(true);
  }, []);

  const getVolumeImage = (volumeInfo: VolumeInfo) => {
    if (!volumeInfo) return;

    return (
      volumeInfo?.imageLinks?.thumbnail ||
      volumeInfo?.imageLinks?.smallThumbnail
    );
  };

  return (
    <>
      <S.Container>
        <S.Content>
          <Filters />
          {volumes.length ? (
            <S.ContentResults>
              {volumes.map(({ id, volumeInfo }) => (
                <S.ContentResultsWrapper key={id}>
                  <S.ContentResultsCover>
                    <img
                      src={getVolumeImage(volumeInfo)}
                      alt={volumeInfo?.title}
                      loading="lazy"
                    />
                  </S.ContentResultsCover>
                  <S.ContentResultsTitle>
                    <label>{volumeInfo?.title} </label>
                  </S.ContentResultsTitle>
                  <S.ContentResultsCategory>
                    <span>{volumeInfo?.authors?.join(", ")}</span>
                  </S.ContentResultsCategory>
                </S.ContentResultsWrapper>
              ))}
            </S.ContentResults>
          ) : (
            <>
              <Spinner />
              <h2>Carregando...</h2>
            </>
          )}
        </S.Content>
      </S.Container>
      <Footer
        text={footerAllrightsReserved}
        privacyText={privacyAndPolicy}
        termsAndUsageText={termsAndUsage}
        helpText={helpLink}
      />
    </>
  );
};

export default Search;
