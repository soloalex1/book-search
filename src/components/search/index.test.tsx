import { describe, expect, test } from "vitest";

import { screen, render } from "@testing-library/react";

import Search from ".";

describe("components/Search", () => {
  test("should render the component correctly", () => {
    const { container } = render(<Search />);

    expect(container).toMatchSnapshot();

    const searchField = screen.getByPlaceholderText(/pesquisar/i);
    expect(searchField).toBeVisible();
    expect(searchField).toBeEnabled();

    const searchIcon = screen.getByLabelText(/pesquisar/i);
    expect(searchIcon).toBeVisible();
  });
});
