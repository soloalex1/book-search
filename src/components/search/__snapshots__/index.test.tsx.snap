// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/Search > should render the component correctly 1`] = `
<div>
  <form
    class="sc-bczSft emRDFI"
  >
    <input
      class="sc-gsnScm bqNyGc"
      name="search"
      placeholder="Pesquisar..."
      type="text"
      value=""
    />
    <svg
      aria-label="Pesquisar"
      fill="none"
      height="16"
      title="Pesquisar"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M11.3546 10.4117C12.1757 9.38528 12.6667 8.08329 12.6667 6.66669C12.6667 3.35316 9.98027 0.666687 6.66675 0.666687C3.35322 0.666687 0.666748 3.35316 0.666748 6.66669C0.666748 9.98021 3.35322 12.6667 6.66675 12.6667C8.08335 12.6667 9.38534 12.1757 10.4118 11.3546L14.1953 15.1381C14.4557 15.3984 14.8778 15.3984 15.1382 15.1381C15.3985 14.8777 15.3985 14.4556 15.1382 14.1953L11.3546 10.4117ZM6.66675 2.00002C9.24389 2.00002 11.3334 4.08954 11.3334 6.66669C11.3334 9.24383 9.24389 11.3334 6.66675 11.3334C4.0896 11.3334 2.00008 9.24383 2.00008 6.66669C2.00008 4.08954 4.0896 2.00002 6.66675 2.00002Z"
        fill="#406A76"
        fill-rule="evenodd"
      />
    </svg>
  </form>
</div>
`;
